zrml/parimutuel/src/lib.rs

```diff
@@ -336,6 +340,19 @@ mod pallet {
             let external_fees = T::ExternalFees::distribute(market_id, base_asset, &who, amount);
             let amount_minus_fees =
                 amount.checked_sub(&external_fees).ok_or(Error::<T>::Unexpected)?;
+
+            let existential_deposit_asset = T::AssetManager::minimum_balance(asset);
+            let existential_deposit_base_asset = T::AssetManager::minimum_balance(base_asset);
+            // TODO: distribute already charges the fees / modifies storage (verify first, write last rule broken)
+            // TODO: mitigate this by using `get_fee` and then `distribute` after the merge with PR https://github.com/zeitgeistpm/zeitgeist/pull/1183
```

Comment:
We don't have to observe verify first/write last where it's not convenient. If you want to be used, it's probably better to add `require_transactional` to the `do_*` functions. Using `get_fee` has it's own issues.

```suggestion
```